src/response-base.js

```diff
@@ -70,10 +73,9 @@ ResponseBase.prototype._setHeaderProperties = function(header) {
   this.type = utils.type(ct);
 
   // params
-  const params = utils.params(ct);
-  for (const key in params) {
-    if (Object.prototype.hasOwnProperty.call(params, key))
-      this[key] = params[key];
+  this.typeParams = utils.params(ct);
+  if ('charset' in this.typeParams) {
+    this.charset = this.typeParams.charset;
   }
```

Comment:
This is the fix.

I don't know if that's security sensitive, but I know other libraries labeled such flaws as a security issue. Especially when the string comes from some place the user doesn't control... like a remote http server :-)